### PR TITLE
Fix #180 & extend Topaz support.

### LINKED ISSUE #180: [1.12.2] Some Materials not being loaded
The Materials from every Mod that registeres it's Items at RegistryEvent.Register<Item> not being available(without forcing the Material).

### PR DESCRIPTION
**Pull request by ToMe25**
Fix BlakeBr0/MysticalAgriculture#180 by moving some initialzing to RegistryEvent.Register<Item> with priority Low.
This needs some Changes on Cucumber to work!
For this to work Cucumbers ModRegistry.registerItems needs to be changed to use `@SubscribeEvent(priority=EventPriority.LOWEST)` and ModRegistry.registerBlocks needs to be changed to something like
```java
	@SubscribeEvent(priority=EventPriority.LOWEST)
	public void registerBlocks(RegistryEvent.Register<Item> event) {
		for (RegistryObject<Block> block : blocks) {
			if (block.get().getRegistryName() == null) {
				block.get().setRegistryName(block.getName());
			}
			ForgeRegistries.BLOCKS.register(block.get());
		}
	}
```
Added support for other Mods Topaz

### CODE REVIEW

**Comment by ToMe25** [comment]
I know it isn't recommended to register Blocks at RegistryEvent.Register<Item> but i haven't found any other Way to fix that.

**Comment by ToMe25** [comment]
Now it would no longer need any Changes to Cucumbers ModRegistry class but instead it would need a class like
```java
package com.blakebr0.cucumber.registry;

import net.minecraft.block.Block;
import net.minecraft.item.Item;
import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.event.RegistryEvent.Register;
import net.minecraftforge.fml.common.eventhandler.EventPriority;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
import net.minecraftforge.fml.common.registry.ForgeRegistries;

public class LateModRegistry extends ModRegistry {

	public LateModRegistry(String modid) {
		super(modid);
	}
	
	public static LateModRegistry create(String modid) {
		LateModRegistry registry = new LateModRegistry(modid);
		MinecraftForge.EVENT_BUS.register(registry);
		return registry;
	}
	
	@Override
	public void registerBlocks(Register<Block> event) {
		//do Nothing!
	}
	
	@SubscribeEvent(priority=EventPriority.LOWEST)
	public void registerBlocksLate(Register<Item> event) {
		for (RegistryObject<Block> block : blocks) {
			if (block.get().getRegistryName() == null) {
				block.get().setRegistryName(block.getName());
			}
			ForgeRegistries.BLOCKS.register(block.get());
		}
	}
	
	@Override
	@SubscribeEvent(priority=EventPriority.LOWEST)
	public void registerItems(Register<Item> event) {
		super.registerItems(event);
	}
	
}
```
in Cucumber.

**Comment by ToMe25** [comment]
Now it should not need any Changes to the Cucumber Library as the LateModRegistry Class is now a part of Mystical Agriculture.

**Comment by ToMe25** [comment]
@BlakeBr0 can you please comment what you are thinking about this

**Comment by ToMe25** [comment]
@BlakeBr0 can you please atleast Comment what you are thinking about this PR

**Comment by BlakeBr0** [comment]
I think the load order stuff is just going to be an issue in this version. I would prefer not to try to hack around it. I added the ability to force enable crops specifically because of this issue and don't really want to touch this anymore. 

In 1.14 and on this won't be an issue anymore.

**Comment by ToMe25** [comment]
I wouldn't see this as "hack around it" i just moved it all to a lower priority to make the checks working,
but if you don't want to use things like this at all and not just a part of it i guess i can't get it work in a less "hacky" way.

I know one kinda stupid thing...
would you use it if i prevent it from registering any Stuff twice?

**Comment by ToMe25** [comment]
I originally didn't added that dummy Methods because at that time i was thinking its even more "hacky" but i do think since some time things like this are "better".

File: src/main/java/com/blakebr0/mysticalagriculture/registry/LateModRegistry.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		MinecraftForge.EVENT_BUS.register(registry);
+		return registry;
+	}
+	
+	public void registerBlocksLate(Register<Item> event) {
+		for (RegistryObject<Block> block : blocks) {
+			if (block.get().getRegistryName() == null) {
+				block.get().setRegistryName(block.getName());
+			}
+			ForgeRegistries.BLOCKS.register(block.get());
```
Comment (ToMe25) [inline src/main/java/com/blakebr0/mysticalagriculture/registry/LateModRegistry.java]: If this is the Point you don't want i could cache the Registry from `RegistryEvent.Registery<Block>` and use that one here, but i don't know any way to make this working with the Block Registry Event.

**Comment by BlakeBr0** [comment]
I don't think I'm going to implement this. I appreciate your help though.

**Pull request closed without merging**